lib/services.js

```diff
@@ -128,12 +128,11 @@ const services = {
         if (lastModifiedDate) {
             md.setLastModified(lastModifiedDate);
         }
-        if (cipherBundle) {
-            md.setAmzServerSideEncryption(cipherBundle.algorithm);
-            // configuredMasterKeyId takes precedence
-            if (cipherBundle.configuredMasterKeyId || cipherBundle.masterKeyId) {
-                md.setAmzEncryptionKeyId(cipherBundle.configuredMasterKeyId || cipherBundle.masterKeyId);
-            }
+
+        if (cipherBundle || params.cipherBundle) {
+            const bundle = cipherBundle || params.cipherBundle;
+            md.setAmzServerSideEncryption(bundle.algorithm);
+            md.setAmzEncryptionKeyId(bundle.configuredMasterKeyId || bundle.masterKeyId);
```

Comment:
should we not keep the test which was present before?
e.g. do not set `AmzEncryptionKeyId` to null-ish value?